### ISSUE TITLE
[Bug] Cannot properly parse custom commands containing quotes

### ISSUE THREAD

**Issue by shukiv** [grafviktor/goto#75]
Hello,

Nice work. But I am missing the ProxyCommand feature.

Thanks!

**Comment by grafviktor** [comment]
Hi @shukiv. Thank you! 🙃 You can define any custom parameters placing them right in the `Host` field (do not prepend your parameters with `ssh` as it will be done automatically). Something like this:

![extra_parameters](https://github.com/user-attachments/assets/e42bcefb-27a7-47b2-a333-6fb5a0fba3ef)

I should have updated the project documentation. Will do that after the next app release.

**Comment by shukiv** [comment]
Hey,

Thanks for your input. But it is not working for me.
I am using this line in the "command" part:
ssh root@123.123.123.123 -o ProxyCommand="nc -x 127.0.0.1:9689 %h %p" -i /home/user/.ssh/id_rsa

But I am getting:
Command: ssh root@123.123.123.123 -o ProxyCommand="nc -x 127.0.0.1:9689 %h %p" -i /home/user/.ssh/id_rsa
Error:   command-line line 0: invalid quotes

I have tried with ', \", but nothing works.

Can you help?

Thanks in advance,
Shuki

**Comment by grafviktor** [comment]
Agree, that becomes trickier when quotes are used. Could you please try to escape them:

```bash
# each quote is escaped with \"
ssh root@123.123.123.123 -o ProxyCommand=\"nc -x 127.0.0.1:9689 %h %p\" -i /home/user/.ssh/id_rsa 
```

If doesn't work still, please provide your OS, shell (bash, zsh, etc...). Will try to simulate this behavior locally.

**Comment by shukiv** [comment]
Hey,

When I use quotes I get this error:
/bin/bash: line 1: exec: "nc: not found                                                                                                                       

I tried to:
ssh root@123.123.123.123 -o ProxyCommand=\"/usr/bin/nc -x 127.0.0.1:9689 %h %p\" -i /home/user/.ssh/id_rsa 

But I am getting:
/bin/bash: line 1: /home/shuki/"/usr/bin/nc: No such file or directory                                                                                        

As you can see it's bash and I am on Debian 12.

Thanks!

**Comment by grafviktor** [comment]
Thank you. Does it work if you run the command below directly in your terminal?

```bash
ssh root@123.123.123.123 -o ProxyCommand="/usr/bin/nc -x 127.0.0.1:9689 %h %p" -i /home/user/.ssh/id_rsa
# try removing full path to netcat as well
ssh root@123.123.123.123 -o ProxyCommand="nc -x 127.0.0.1:9689 %h %p" -i /home/user/.ssh/id_rsa
```

I will try to run Debian 12 in Docker and reproduce the problem in the meanwhile.

**Comment by shukiv** [comment]
I tried with and without full path. Nothing works.

**Comment by shukiv** [comment]
And yes, it works on the terminal.

**Comment by grafviktor** [comment]
> And yes, it works on the terminal.
Confirming. I can reproduce the problem. Will try to find a workaround and update you. Will raise a bug to cover that.

**Comment by shukiv** [comment]
Thank you so much!
Thanks again for this nice program.

**Comment by grafviktor** [comment]
### Analysis:

Utility [`BuildProcess`](https://github.com/grafviktor/goto/blob/c4e0eb7b6364d80c24de28e4985e403b77b578ae/internal/utils/utils.go#L95) function splits ssh parameters into an array for rurther forwarding those params into standard library's [`os.Exec`](https://pkg.go.dev/os/exec#Command) call. `space` character is used as a separator for splitting arguments.

However, this approach fails when ssh parameters contain subcommands which are provided in quotes. This set of parameters reveals the problem:
```bash
ssh roman@127.0.0.1 -o ProxyCommand="/usr/bin/nc -x 127.0.0.1:9689 %h %p" -i /Users/roman/.ssh/id_rsa
``` 
![debugger](https://github.com/user-attachments/assets/789bba0d-fee2-414c-9892-69c7f5ba44ba)

Because of the extra quotes `os.Exec` function fails.

**Comment by grafviktor** [comment]
@shukiv , thank you very much for this bug report ! 

If you have a couple of minutes, could you run a test build from the attachment on your machine and confirm whether the problem still exists? Do not escape quotes this time. Please note this build is unstable and some functionality can be broken.

[goto-v1.2.0.zip](https://github.com/user-attachments/files/16203722/goto-v1.2.0.zip)

**Comment by shukiv** [comment]
It works! Amazing! Thank you Roman!

**Comment by grafviktor** [comment]
> It works! Amazing! Thank you Roman!

Shuki, thanks very much for your help! The fix will be included in release `1.2.0` which will be available within 2-3 weeks from now 🙂
